docs/Streams/index.md

```diff
@@ -0,0 +1,73 @@
+# Streams
+
+A stream is a doubly linked list of bytes. It can be used for file input/output, DayZ serializers, hex conversion or string parsing. Once a stream is finished, it is recommended to close the stream either by letting the reference count equal zero or call the `Close` method.
+
+Streams can be manipulated with `CF_TextWriter`, and `CF_BinaryWriter`. Streams can be read with `CF_TextReader`, and`CF_BinaryReader`.
+
+## Files
+
+Use the `CF_FileStream` class to read and write files on the a file system. 
+
+```csharp
+string path = "$profile:test.txt";
+
+if (FileExist(path)) DeleteFile(path);
+
+// Create the file
+CF_TextWriter writer = new CF_TextWriter(new CF_FileStream(path, FileMode.WRITE));
+writer.WriteLine("This is some text");
+writer.Close();
+
+// Open the file and read it back
+CF_TextReader reader = new CF_TextReader(new CF_FileStream(path, FileMode.READ));
+Print(reader.ReadLine()); // "This is some text")
+reader.Close();
+```
+
+## String
+
+Use the `CF_StringStream` class to read and write to a string source. This stream does not accept '\0' bytes.
+
+```csharp
+string data = "Hello, World!";
+
+CF_TextReader reader = new CF_TextReader(new CF_StringStream(data));
+Print("ReadWord: " + reader.ReadWord());				// "Hello"
+Print("ReadChar: " + reader.ReadChar());				// ","
+Print("ReadWhitespace: " + reader.ReadWhitespace());	// ""
+Print("ReadWord: " + reader.ReadWord());				// "World"
+Print("ReadChar: " + reader.ReadChar());				// "!"
+reader.Close();
+```
+
+## Hex
+
+Use the `CF_HexStream` class to read and write to a string source as a hex output.
+
+```csharp
+CF_HexStream stream = new CF_HexStream();
+
+CF_BinaryWriter writer = new CF_BinaryWriter(stream);
+writer.WriteFloat(5);
+writer.Close();
+
+Print(stream.ToStr());	// "40A00000"
```

Comment:
If you make an example about float, maybe append the endianness and float format to the comment.